lib/Sema/CSSimplify.cpp

```diff
@@ -3327,6 +3327,10 @@ ConstraintSystem::simplifyBridgingConstraint(Type type1,
                                              Type type2,
                                              TypeMatchOptions flags,
                                              ConstraintLocatorBuilder locator) {
+  // There's no bridging without ObjC interop.
+  if (!TC.Context.LangOpts.EnableObjCInterop)
```

Comment:
That might be a good cleanup for master, but I don't think it makes sense to take the risk in 3.1.